examples/core/quad/main.rs

```diff
@@ -317,7 +331,8 @@ fn main() {
     let upload_type =
         memory_types.iter().find(|mem_type| {
             buffer_req.type_mask & (1 << mem_type.id) != 0 &&
-            mem_type.properties.contains(m::CPU_VISIBLE | m::COHERENT)
+            mem_type.properties.contains(m::CPU_VISIBLE) &&
+            mem_type.properties.intersects(m::CPU_CACHED | m::WRITE_COMBINED)
```

Comment:
> There must be at least one memory type with both the VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT and
VK_MEMORY_PROPERTY_HOST_COHERENT_BIT bits set in its propertyFlags. There must be at least one
memory type with the VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT bit set in its propertyFlags.

We might end up without an upload heap here.

Reply:
What do you propose? Use `COHERENT` here as a fallback?